nf/CELLRANGER/main.nf

```diff
@@ -2,28 +2,43 @@
 
 include { COUNT } from './modules/local/cell_ranger/count/main.nf'
 include { MULTI } from './modules/local/cell_ranger/multi/main.nf'
-include { UNTAR_REF } from './modules/local/tar/main.nf'
+include { UNTAR_REF } from './modules/local/untar_ref/main.nf'
+include { TAR_DIR } from './modules/local/tar_dir/main.nf'
+
+def validate_inputs(param_obj){
+    if (!param_obj.transcriptome_tar && !param_obj.transcriptome_dir) {
+        error "Must provide one of either a path to a transcriptome directory or a tar file of the reference!"
+    }
+    def required_params = ["mode", "reads", "mates", "create_bam"]
+    def required_count_params = ["sample"]
+    def required_multi_params = ["library_fastq_id", "sample_csv", "feature_types"]
+    required_params.each{ param -> if (!param_obj[param]) error "Missing required parameter: ${param}" }
+    if (param_obj.mode == "count"){
+        required_count_params.each{ param -> if (!param_obj[param]) error "Missing required parameter: ${param}" }
+    }
+    if (param_obj.mode == "multi"){
+        required_multi_params.each{ param -> if (!param_obj[param]) error "Missing required parameter: ${param}" }
+    }
```

Comment:
Rather than throwing the error whenever you encounter something missing, maybe you do a flag + collection type deal where it will show all of the missing parameters?